src/main/java/com/project/controller/ApplicationController.java

```diff
@@ -29,40 +29,24 @@ public ApplicationController(ApplicationService applicationService) {
     }
 
     @GetMapping("api/applications/{id}")
-    public ResponseEntity<Application> getApplication(@PathVariable long id) {
-        try {
-            return new ResponseEntity<Application>(applicationService.getById(id), HttpStatus.OK);
-        } catch (ApplicationNotFoundException e) {
-            return ResponseEntity.notFound().build();
-        }
+    public ResponseEntity<Application> getApplication(@PathVariable long id) throws ApplicationNotFoundException {
+        return new ResponseEntity<Application>(applicationService.getById(id), HttpStatus.OK);
     }
 
     @PostMapping("api/applications")
-    ResponseEntity<Application> postUser(@Valid @RequestBody Application newApplication) throws UserException {
+    ResponseEntity<Application> postApplication(@Valid @RequestBody Application newApplication) throws UserException, ApplicationNotFoundException {
         applicationService.addApplication(newApplication);
-        try {
-            return new ResponseEntity<Application>(applicationService.getById(newApplication.getId()), HttpStatus.CREATED);
-        } catch (ApplicationNotFoundException e) {
-            return ResponseEntity.badRequest().build();
-        }
+        return new ResponseEntity<Application>(applicationService.getById(newApplication.getId()), HttpStatus.CREATED);
```

Comment:
You could return saved application from applicationService, since CrudRepository method save returns the objecct you saved. Wouldn't need one additional call to database then.